src/transformers/models/pix2struct/image_processing_pix2struct.py

```diff
@@ -282,6 +282,17 @@ def render_header(self, image: ImageInput, header: str, **kwargs):
         """
         requires_backends(self.render_header, "vision")
 
+        return_framework = None
+
+        # Convert to PIL image if necessary
+        if not isinstance(image, Image.Image):
```

Comment:
There's a `to_pil_image`  and `to_numpy_array` you should use here instead which handles things like any necessary type conversions, axis permutations and checking if the libraries e.g. torch are in the environment.

Reply:
Thanks! Fixed in [17bae00](https://github.com/huggingface/transformers/pull/21400/commits/17bae00a34ded765e671beb8dfba10b2fdd32e3a)